### ISSUE TITLE
Incorrectly warning about ignored files

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#3649]
After globs were added to the cli input in 5a599c70a98eaed0d43a8f28690921ef4ad0702d, my command to lint rule docs has broken:

```
DEBUG=eslint* node_modules/.bin/eslint --plugin markdown --no-eslintrc --config conf/rule-doc-config.json docs/rules/quote-props.md
  eslint:cli Running on files +0ms
  eslint:config Using command line config conf/rule-doc-config.json +3ms
  eslint:cli-engine Linting complete in: 3ms +10ms

/home/ian/Local_code/eslint/eslint/docs/rules/quote-props.md
  0:0  warning  File ignored because of your .eslintignore file. Use --no-ignore to override

✖ 1 problem (0 errors, 1 warning)
```

Compare to my output if I revert to the commit prior to globs being added:

```
▶ git reset --hard 37bb3159f748463861ba9aadf704e76af3c2537d
HEAD is now at 37bb315 Merge pull request #3518 from ev0rtex/update-expression-rule-logic

▶ DEBUG=eslint* node_modules/.bin/eslint --plugin markdown --no-eslintrc --config conf/rule-doc-config.json docs/rules/quote-props.md
  eslint:cli Running on files +0ms
  eslint:config Using command line config conf/rule-doc-config.json +3ms
  eslint:cli-engine Processing docs/rules/quote-props.md +10ms
  eslint:cli-engine Linting docs/rules/quote-props.md +1ms
  eslint:config Constructing config for /home/ian/Local_code/eslint/eslint/docs/rules/quote-props.md +0ms
  eslint:config Not using .eslintrc or package.json files +0ms
  eslint:config Merging command line config file +1ms
  eslint:config Merging command line environment settings +0ms
  eslint:config Merging command line plugins +0ms
  eslint:cli-engine Load plugin markdown +4ms

docs/rules/quote-props.md
    7:20  error  Unexpected token ;
  130:6   error  Unexpected string

✖ 2 problems (2 errors, 0 warnings)
```

Something appears to have changed in how files are determined to be ignored.  Furthermore, if I follow the warning's advice and add `--no-ignore`, the warning is not shown, but the file is still not linted:

```
▶ DEBUG=eslint* node_modules/.bin/eslint --plugin markdown --no-ignore --no-eslintrc --config conf/rule-doc-config.json docs/rules/quote-props.md
  eslint:cli Running on files +0ms
  eslint:config Using command line config conf/rule-doc-config.json +2ms
  eslint:cli-engine Linting complete in: 2ms +9ms
```

**Comment by gyandeeps** [comment]
I am lil confused with the path you provide to run eslint which is from node_modules folder but all the other paths like eslintignore file path come from a different location.

**Comment by IanVS** [comment]
I'm running this from inside the eslint project.  I don't have ESLint installed globally, and with the `eslint-plugin-markdown` that I'm using, I've had to install eslint within eslint (which I do with an `npm link` and `npm link eslint`).  It's a little unorthodox, I know, but it was working just fine.  I could also have done:

```
DEBUG=eslint* node bin/eslint.js --plugin markdown --no-eslintrc --config conf/rule-doc-config.json docs/rules/quote-props.md            
  eslint:cli Running on files +0ms
  eslint:config Using command line config conf/rule-doc-config.json +3ms
  eslint:cli-engine Linting complete in: 3ms +11ms

/home/ian/Local_code/eslint/eslint/docs/rules/quote-props.md
  0:0  warning  File ignored because of your .eslintignore file. Use --no-ignore to override

✖ 1 problem (0 errors, 1 warning)
```

There is no eslintignore path being specified, so I'm not sure what you mean there.

**Comment by IanVS** [comment]
Oh, and if you're wondering, `conf/rule-doc-config.json` is:

```json
{
    "root": true,
    "env": {
        "es6": true
    },
    "ecmaFeatures": {
        "modules": true
    }
}
```

**Comment by ilyavolodin** [comment]
This looks like a bug. On top of that, we should decorate code that loads `.eslintignore` with debug statements, so we could trace them.

**Comment by IanVS** [comment]
@BYK Do you think you could take a look at this?  It seems you did quite a few changes to the ignore logic when you implemented globs.

**Comment by gyandeeps** [comment]
@IanVS Can you share the content of `.eslintignore` . thanks

**Comment by IanVS** [comment]
There's not one.  I'm linting the eslint rule docs.

**Comment by gyandeeps** [comment]
Here is the answer, Becausse by default eslint looks for file with extensions `.js`. Now if you say `--ext .md` then it would work fine.

**Comment by IanVS** [comment]
Previously, if you specified a full filename, it would be linted no matter the extension, which I believe is correct.  The `--ext` flag should only be needed when you are providing a directory to search, IMHO.

**Comment by nzakas** [comment]
@ianvs is correct - we should lint anything that is passed explicitly on the command line.
